### DISCUSSION TITLE
Compress MCAP file with C++ library code.

### DISCUSSION THREAD

**Discussion by yun-li** [foxglove/mcap#1132] [General] [answered]
I used C++ library to generate MCAP file, the file ending up being huge and can be compress with `mcap compress` command. Is there a way to do this compress use C++ code?

**Comment by jtbandes** [comment]
Duplicate of https://github.com/orgs/foxglove/discussions/919

**Comment by defunctzombie** [comment, marked as answer]
@yun-li You can provide writer options: https://mcap.dev/docs/cpp/r832FE362A16BB6E8 to the writer to enable compression.
